erpnext/accounts/report/account_group_mapping/account_group_mapping.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) 2025, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe import _
+
+def execute(filters=None):
+	return AccountGroupMappingReport(filters).run()
+
+class AccountGroupMappingReport:
+	def __init__(self, filters=None):
+		self.filters = frappe._dict(filters or {})
+		self.company = self.filters.get("company")
+		self.report_type = self.filters.get("report_type")
+		self.root_type = self.filters.get("root_type")
+
+	def run(self):
+		self.validate_filters()
+		return self.get_columns(), self.get_data()
+
+	def validate_filters(self):
+		if not self.company:
+			frappe.throw(_("Company filter is required"))
+
+	def get_account_groups(self):
+		filters = {"company": self.company, "report_type": self.report_type}
+		return frappe.get_all(
+			"Account Group",
+			filters=filters,
+			fields=["name", "group_name", "report_type", "root_type"],
+			order_by="group_name"
+		)
+
+	def get_leaf_accounts(self):
+		conditions = ["company=%s", "rgt - lft = 1", "report_type=%s"]
+		args = [self.company, self.report_type]
```

Comment:
rather than building your own args, you can simply pass self.filters since all the value would be there already